src/grt/src/GlobalRouter.cpp

```diff
@@ -785,8 +785,16 @@ void GlobalRouter::initNets(std::vector<Net*>& nets)
   }
 
   if (critical_nets_percentage_ != 0) {
-    fastroute_->setUpdateSlack(critical_nets_percentage_);
-    computeNetSlacks();
+    if (sta_->getDbNetwork()->defaultLibertyLibrary() != nullptr) {
+      fastroute_->setUpdateSlack(critical_nets_percentage_);
+      computeNetSlacks();
+    } else {
```

Comment:
I think you could also add a flag to the `initNets` function to detect if it's being called during `readGuides`/`loadGuidesFromDB` or during the normal grt flow. Currently, the warning message can be generated when using the `read_guides` command, which I think is not correct.